vowpalwabbit/example.cc

```diff
@@ -52,6 +55,10 @@ example::example(example&& other) noexcept
     , in_use(other.in_use)
 {
   other.weight = 1.f;
+  auto& other_cats_pdf = other.meta_info.cats_pdf_hint;
```

Comment:
It would be more useful to overload the move operator in v_array and just `std::move()` this

Reply:
good point

Reply:
actually I found this comment in v_array and I am guessing it holds for move constructors too

```
// v_array cannot have a user-defined constructor, because it participates in various unions.
  // union members cannot have user-defined constructors.
  // v_array() : _begin(nullptr), _end(nullptr), end_array(nullptr), erase_count(0) {}
```

Reply:
ah you said move operator, never mind

Reply:
actually, can't add move assignment operator either

copy constructors are not auto generated if you specify a move assignment operator
if copy constructor is non-trivial then you can't use the class in a union

Reply:
One day we can add them... :/ Need those unions gone first...